### PR TITLE
Stacks 2.1

### PR DESCRIPTION
**Pull request by jcnelson**
Stacks 2.1 stable release:  Our love letter to the Stacks community on Valentine's Day 2023.

### CODE REVIEW

**Comment by jcnelson** [comment]
Still TODO:
* Pick the block height at which 2.1 goes live
* Update SIP-015 to have this block-height

**Comment by diwakergupta** [comment]
> * Pick the block height at which 2.1 goes live, for both testnet and mainnet

Can we align on the high-level timeline for mainnet? Right now Stacking cycle 52 is in progress. 
* Cycle 53 begins in approx 5 days (block 777250)
* Rough expectation is activation ~2 cycles after binaries are available
* So I would imagine activation is set for end of cycle 54, roughly block 781450

I heard someone mention end of cycle 55 (which would be in April), which seems farther out than necessary.

cc/ @cuevasm @wileyj

**Comment by igorsyl** [comment]
@diwakergupta I shared a calendar invite with you to communicate the consensus on the target activation date.

**Comment by igorsyl** [comment]
@diwakergupta I shared a calendar invite with you to communicate the consensus on the target activation date.

**Comment by diwakergupta** [comment]
> @diwakergupta I shared a calendar invite with you to communicate the consensus on the target activation date.

Yep, thanks. But that invite factors in _three_ cycles. The expectation always has been _two_ cycles. So my question is, why are we adding an extra two weeks for activation. 2.1 is so far delayed as-is, IMO there's no reason to delay activation any further than necessary.

**Comment by jcnelson** [comment]
> I heard someone mention end of cycle 55 (which would be in April), which seems farther out than necessary.

My recommendation was the _start_ of cycle 55

**Comment by diwakergupta** [comment]
> > I heard someone mention end of cycle 55 (which would be in April), which seems farther out than necessary.
> 
> My recommendation was the _start_ of cycle 55

Great, that seems in line with what I had posted above. Roughly what block number did you have in mind?

@igorsyl perhaps you can update your invite to avoid confusion? Specifically change "Estimated activation in 6 days + 3 cycles x 14 days/cycle = 48 days => April 3" to use 2 cycles instead.

**Comment by jcnelson** [comment]
> Great, that seems in line with what I had posted above. Roughly what block number did you have in mind?

The start of cycle 55 is Bitcoin block 781,551

**Comment by igorsyl** [comment]
**Updated from END of Cycle 55 to START of Cycle 55.**

[Cycle 52](https://www.google.com/url?q=https://stacking.club/cycles/52&sa=D&source=calendar&ust=1676838493294830&usg=AOvVaw1iO3TBxICi-oGEhv0Zb8V7) ends in ~6 days.
Estimated activation in 6 days + 2 cycles x 14 days/cycle = 34 days => March 20

cc @diwakergupta

**Comment by igorsyl** [comment]
Failed test cases:

```
[sampled-genesis (tests::neon_integrations::stx_transfer_btc_integration_test)](https://github.com/stacks-network/stacks-blockchain/actions/runs/4175444795/jobs/7230824324#logs)

#5 181.3 thread 'tests::neon_integrations::stx_transfer_btc_integration_test' panicked at 'unexpected btc transaction type', testnet/stacks-node/src/tests/neon_integrations.rs:1706:21
```

```
[sampled-genesis (tests::epoch_21::test_v1_unlock_height_with_current_stackers)](https://github.com/stacks-network/stacks-blockchain/actions/runs/4175444795/jobs/7230835391#logs)

#5 215.2 thread 'tests::epoch_21::test_v1_unlock_height_with_current_stackers' panicked at 'assertion failed: `(left == right)`
#5 215.2   left: `Tuple(TupleData { type_signature: TupleTypeSignature { "hashbytes": (buff 20), "version": (buff 1),}, data_map: {ClarityName("hashbytes"): Sequence(Buffer(0000000000000000000000000000000000000000)), ClarityName("version"): Sequence(Buffer(00))} })`,
#5 215.2  right: `Tuple(TupleData { type_signature: TupleTypeSignature { "hashbytes": (buff 20), "version": (buff 1),}, data_map: {ClarityName("hashbytes"): Sequence(Buffer(0e6689f4efced11c29b55c4a9c33b67a82758cab)), ClarityName("version"): Sequence(Buffer(00))} })`', testnet/stacks-node/src/tests/epoch_21.rs:5138:21
```

**Comment by jcnelson** [comment]
Should be fixed in https://github.com/stacks-network/stacks-blockchain/pull/3572.

Once that PR merges, we can merge this one.

File: src/chainstate/stacks/block.rs [hunk 1]
```diff
@@ -1010,69 +1048,6 @@ mod test {
 
     #[test]
     fn codec_stacks_block() {
-        /*
```
Comment (pavitthrap) [inline src/chainstate/stacks/block.rs:1013]: delete

File: src/net/chat.rs [hunk 2]
```diff
@@ -1243,6 +1244,7 @@ impl ConversationP2P {
         chain_view: &BurnchainView,
         message: &mut StacksMessage,
     ) -> Result<Option<StacksMessage>, net_error> {
+        // monitoring::increment_p2p_msg_ping_received_counter();
```
Comment (pavitthrap) [inline src/net/chat.rs:1247]: delete

File: src/net/chat.rs [hunk 3]
```diff
@@ -1266,6 +1268,7 @@ impl ConversationP2P {
         chain_view: &BurnchainView,
         preamble: &Preamble,
     ) -> Result<ReplyHandleP2P, net_error> {
+        // monitoring::increment_p2p_msg_get_neighbors_received_counter();
```
Comment (pavitthrap) [inline src/net/chat.rs:1271]: delete

**Review by pavitthrap — APPROVED** [review]
just had a few comments about dead code

**Review by kantai — APPROVED** [review]
LGTM!

**Review by igorsyl — APPROVED** [review]
image

**Comment by jcnelson** [comment]
Just gonna write a changelog and address @pavitthrap's feedback real quick (in a separate PR).

**Pull request merged**